coreTable/OwidTable.ts

```diff
@@ -287,10 +287,12 @@ export class OwidTable extends CoreTable<OwidRow, OwidColumnDef> {
         const newDef = {
             ...col.def,
             type: ColumnTypeNames.Percentage,
-            values: col.parsedValues.map(
-                (val, index) =>
-                    (100 * (val as number)) / timeTotals.get(timeValues[index])!
-            ),
+            values: col.parsedValues.map((val, index) => {
+                const timeTotal = timeTotals.get(timeValues[index])
+                if (timeTotal === undefined || timeTotal === 0)
```

Comment:
Hmm, timeTotal should never be undefined right? And the code proves that (at runtime anyway). I understand that when using a map TypeScript will give us that possibility, and that non-null assertions are bad, but is there a better way we could do this?

TypeScript gives an error for this, that `map.get(1)` is possibly undefined, even though we can easily see that it can never be undefined. So in this case better to use a non-null assertion than to handle the undefined cases?
```
const map = new Map<number, string>() 
map.set(1, "a")
const letterA = map.get(1).substr(0, 1)
```

Reply:
As you probably know by now, I like to play it safe and explicitly check for undefined, whereas you reach for type assertions more often :)

Yes, I think in this case we can use the type assertion. The risk here, as always, is when someone changes `sumsByTime` so it no longer returns the correct values in some case, and then this assumption that we have in our minds (but not in the type system!) no longer holds.
But there are many, many more (and more likely!) cases where this can happen, especially with array indices.

Reply:
Ah, that's a very good point. This is a good thread about this type of situation: https://github.com/microsoft/TypeScript/issues/9619

Reply:
But it does not seem like there is a solution yet